src/Plugins/OrientationAnalysis/src/OrientationAnalysis/utilities/TiffWriter.cpp

```diff
@@ -0,0 +1,175 @@
+#include "TiffWriter.hpp"
+
+#include <array>
+#include <cstddef>
+#include <fstream>
+#include <vector>
+
+namespace
+{
+
+const uint16_t PHOTOMETRIC_MINISBLACK = 0x0001;
+const uint16_t PHOTOMETRIC_RGB = 0x0002;
+
+struct TIFTAG
+{
+  int16_t tagId = 0;      // The tag identifier
+  int16_t dataType = 0;   // The scalar type of the data items
+  int32_t dataCount = 0;  // The number of items in the tag data
+  int32_t dataOffset = 0; // The byte offset to the data items
+
+  friend std::ostream& operator<<(std::ostream& out, const TIFTAG& tag)
+  {
+    out.write(reinterpret_cast<const char*>(&tag.tagId), sizeof(tag.tagId));
+    out.write(reinterpret_cast<const char*>(&tag.dataType), sizeof(tag.dataType));
+    out.write(reinterpret_cast<const char*>(&tag.dataCount), sizeof(tag.dataCount));
+    out.write(reinterpret_cast<const char*>(&tag.dataOffset), sizeof(tag.dataOffset));
+
+    return out;
+  }
+};
+
+enum class Endianess
+{
+  Little = 0,
+  Big
+};
```

Comment:
```suggestion
```
There is no reason to redefine this. It is already defined in the `Bit.hpp` file, under the enum `endian`

Reply:
Done